lib/samples/snap_geometry_edits/snap_geometry_edits.dart

```diff
@@ -0,0 +1,679 @@
+                    _polylineLayerSnapSources,
+                  ),
+                  // Add checkboxes for the graphics overlay as snap sources.
+                  buildSnapSourcesSelection(
+                    'Graphics Overlay',
+                    _graphicsOverlaySnapSources,
+                  ),
+                ],
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget buildSnapSourcesSelection(
+    String label,
+    List<SnapSourceSettings> allSourceSettings,
+  ) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(
+          label,
+          style: Theme.of(context).textTheme.titleMedium,
+        ),
+        Column(
+          children: allSourceSettings.map((sourceSetting) {
+            return Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                // Display the layer name, or set default text for graphics overlay.
+                Text(
+                  allSourceSettings == _pointLayerSnapSources ||
+                          allSourceSettings == _polylineLayerSnapSources
+                      ? (sourceSetting.source as FeatureLayer).name
+                      : 'Editor Graphics Overlay',
+                ),
+                // A checkbox to toggle whether this source setting is enabled.
+                Checkbox(
+                  value: sourceSetting.isEnabled,
+                  onChanged: (isEnabled) {
+                    if (isEnabled != null) {
+                      setState(() => sourceSetting.isEnabled = isEnabled);
+                    }
+                  },
+                ),
+              ],
+            );
+          }).toList(),
+        ),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text('Enable all'),
+            // A checkbox to enable all source settings in the category.
+            Checkbox(
+              value: allSourceSettings.firstWhereOrNull(
+                    (snapSourceSettings) =>
+                        snapSourceSettings.isEnabled == false,
+                  ) ==
+                  null,
+              onChanged: (allEnabled) {
+                if (allEnabled != null) {
+                  allSourceSettings
+                      .map(
+                        (snapSourceSettings) =>
+                            snapSourceSettings.isEnabled = allEnabled,
+                      )
+                      .toList();
+                  // We must redraw to update the checkbox UI.
+                  setState(() {});
+                }
+              },
+            ),
+          ],
+        ),
+        const SizedBox(height: 20),
+      ],
+    );
+  }
+
+  List<DropdownMenuItem<GeometryType>> configureGeometryTypeMenuItems() {
+    // Returns a list of drop down menu items for each geometry type.
+    return _geometryTypes.map((type) {
+      return DropdownMenuItem(
```

Comment:
I think I would use the `=>` notation here instead of `{ ... }` since it's only executing a single statement. Removes the need for the `return`.